dbms/src/AggregateFunctions/AggregateFunctionAvgWeighted.cpp

```diff
@@ -0,0 +1,47 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/AggregateFunctionAvgWeighted.h>
+#include <AggregateFunctions/Helpers.h>
+#include <AggregateFunctions/FactoryHelpers.h>
+
+namespace DB
+{
+
+namespace
+{
+
+template <typename T>
+struct AvgWeighted
+{
+    using FieldType = std::conditional_t<IsDecimalNumber<T>, Decimal128, NearestFieldType<T>>;
+    using Function = AggregateFunctionAvgWeighted<T, AggregateFunctionAvgWeightedData<FieldType>>;
+};
+
+template <typename T>
+using AggregateFuncAvgWeighted = typename AvgWeighted<T>::Function;
+
+AggregateFunctionPtr createAggregateFunctionAvgWeighted(const std::string & name, const DataTypes & argument_types, const Array & parameters)
+{
+    assertNoParameters(name, parameters);
+    assertBinary(name, argument_types);
+
+    AggregateFunctionPtr res;
+    DataTypePtr data_type = argument_types[0];
+    if (isDecimal(data_type))
+        res.reset(createWithDecimalType<AggregateFuncAvgWeighted>(*data_type, *data_type, argument_types));
+    else
+        res.reset(createWithNumericType<AggregateFuncAvgWeighted>(*data_type, argument_types));
+
+    if (!res)
+        throw Exception("Illegal type " + argument_types[0]->getName() + " of argument for aggregate function " + name,
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+    return res;
+}
+
+}
+
+void registerAggregateFunctionAvgWeighted(AggregateFunctionFactory & factory)
+{
+    factory.registerFunction("AvgWeighted", createAggregateFunctionAvgWeighted, AggregateFunctionFactory::CaseInsensitive);
```

Comment:
It's better to use `AvgWeighted` name by our style.
Also I don't think that it should be CaseInsensitive, cause `AVGWEIGHTED` looks wired. But can leave it if you want.

Reply:
How?
I looked here https://github.com/ClickHouse/ClickHouse/issues/2014#issuecomment-372445776

Reply:
`avgWeighted` for sure.
Because we are using camelCase (starting from lowercase letter except if it is starting by all caps acronym).